builtin/providers/aws/resource_aws_api_gateway_method_response.go

```diff
@@ -0,0 +1,207 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayMethodResponse() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayMethodResponseCreate,
+		Read:   resourceAwsApiGatewayMethodResponseRead,
+		Update: resourceAwsApiGatewayMethodResponseUpdate,
+		Delete: resourceAwsApiGatewayMethodResponseDelete,
+
+		Schema: map[string]*schema.Schema{
+			"api_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"http_method": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"status_code": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"response_models": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayMethodResponseCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	models := make(map[string]string)
+	for k, v := range d.Get("response_models").(map[string]interface{}) {
+		models[k] = v.(string)
+	}
+
+	_, err := conn.PutMethodResponse(&apigateway.PutMethodResponseInput{
+		HttpMethod:         aws.String(d.Get("http_method").(string)),
+		ResourceId:         aws.String(d.Get("resource_id").(string)),
+		RestApiId:          aws.String(d.Get("api_id").(string)),
+		StatusCode:         aws.String(d.Get("status_code").(string)),
+		ResponseModels:     aws.StringMap(models),
+		ResponseParameters: nil,
```

Comment:
Why are we forcing this to be `nil` and not letting the user to specify this via schema/DSL?